src/Ionide.ProjInfo/Library.fs

```diff
@@ -1349,10 +1433,19 @@ type WorkspaceLoader private (toolsPath: ToolsPath, ?globalProperties: (string *
                     let newList, toTrigger =
                         if cache.ContainsKey p then
                             let project = cache.[p]
-                            loadingNotification.Trigger(WorkspaceProjectState.Loaded(project, getAllKnown (), true)) //TODO: Should it even notify here?
+
+                            match project with
+                            | ProjectLoader.LoadedProjectInfo.StandardProjectInfo p -> loadingNotification.Trigger(WorkspaceProjectState.Loaded(p, getAllKnown (), true))
+
+                            | ProjectLoader.LoadedProjectInfo.TraversalProjectInfo p -> ()
+
+                            let referencedProjects =
+                                match project with
+                                | ProjectLoader.LoadedProjectInfo.StandardProjectInfo p -> p.ReferencedProjects
+                                | ProjectLoader.LoadedProjectInfo.TraversalProjectInfo p -> p
```

Comment:
similarly for the standard loader, once we load the outer project if it was a traversal project we can just walk its ProjectReferences instead of creating an entirely new ProjectOptions structure to hold the `.ReferencedProjects` member.